receipt_dynamo/receipt_dynamo/data/base_operations/flattened_mixin.py

```
@@ -117,9 +137,98 @@ def _validate_pagination_params(
+
+        Raises:
+            EntityValidationError: If validation fails
+        """
+        if indices is None:
+            raise EntityValidationError(f"{param_name} cannot be None")
+        if not isinstance(indices, list):
+            raise EntityValidationError(f"{param_name} must be a list")
+        if not all(isinstance(index, tuple) for index in indices):
+            raise EntityValidationError(
+                f"{param_name} must be a list of tuples"
+            )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)